DoorGate/DoorEntity.swift

```diff
@@ -0,0 +1,12 @@
+//
+//  DoorEntity.swift
+//  DoorGate
+//
+//  Created by Yuliani Noriega on 1/29/18.
+//  Copyright © 2018 Yuliani Noriega. All rights reserved.
+//
+import RxSwift
+
+struct DoorEntity {
+    var state = Variable(DoorState.empty.rawValue)
```

Comment:
Bear with me as this will likely be my intro to a lot of things swift...  this Variable() syntax isn't how you typically new a var, right? What is this doing here?

Reply:
Variable() is RXSwift related, I believe its a way to create observable variables in RXSwift. You normally create variable as var name = CLASSTYPEHERE()